src/util-prefilter.h

```diff
@@ -61,25 +61,24 @@ int PrefilterAddSidsResize(PrefilterRuleStore *pmq, uint32_t new_size);
 static inline void PrefilterAddSids(
         PrefilterRuleStore *pmq, const SigIntId *sids, uint32_t sids_size)
 {
-    if (sids_size == 0)
-        return;
-
-    uint32_t new_size = pmq->rule_id_array_cnt + sids_size;
-    if (new_size > pmq->rule_id_array_size) {
-        if (PrefilterAddSidsResize(pmq, new_size) == 0) {
-            // Failed to allocate larger memory for all the SIDS, but
-            // keep as many as we can.
-            sids_size = pmq->rule_id_array_size - pmq->rule_id_array_cnt;
+    if (sids_size > 0) {
```

Comment:
Where do you find the data to support this kind of commit about branch prediction ?

What about `if (UNLIKELY(sids_size == 0))` ?

Reply:
Running things in a profiler like vtune. I'm less confident about using the likely/unlikely logic, I've often seen it not be used. I think it's a hint to the compiler, but it can ignore it, and it seems it often does.

Reply:
So, fo you have a vtune report about all the branches where the predictions are 100% wrong and happen often ?